### DISCUSSION TITLE
multiple agents

### DISCUSSION THREAD

**Discussion by roccho-dev** [plandex-ai/plandex#206] [Q&A] [answered]
Is there a mention at the top of the page regarding multi-agents?
Is this specific use described in the "Background Tasks" section?

https://docs.plandex.ai/intro/

**Comment by danenania** [comment]
Hi @PorcoRosso85! Many/most Plandex operations are using multiple agents in parallel, each with a different role. So just a single task in Plandex can be referred to as "multi-agent". And then on top of that, yes, you can have multiple tasks running in parallel as well, either in the foreground (with an active terminal window that you're streaming to) or in the background on the server.

**Reply by roccho-dev** [reply]
Thanks.

Do you mean to say that the agents you are telling us about can be preconfigured with their respective roles and contexts?

**Reply by danenania** [reply, marked as answer]
No, their roles, prompts, and behavior are configured specifically for Plandex. Here's a list of them: https://docs.plandex.ai/models/roles

**Reply by roccho-dev** [reply]
Thanks for your reply!

- roles already are defined
- roles will be added model which will be used

Any plan of feature like openai/swarm is?

**Reply by danenania** [reply]
Sure thing!

> Any plan of feature like openai/swarm is?

Not currently.

**Reply by roccho-dev** [reply]
One more thing. I need to know if each plandex does not interfere.

If some 'plandex' runs in sepalated session,,,
- each plandex runs independently?
- each plandex communicate only through backend (like cloud or postgres server) ?
